### DISCUSSION TITLE
plugdata as midi effect in Ableton Live 11

### DISCUSSION THREAD

**Discussion by ericbhs** [plugdata-team/plugdata#935] [Q&A]
Hello,

I'd like to use plugdata in Ableton Live 11 as a midi effect (midi in -> midi out). However it seems that the default behaviour is midi in -> audio out in Live. How can I configure it to be midi out ?

I saw that in the midi objects help there is the following note: "In order to use MIDI Input or output objects, be sure to set MIDI devices in: Preferences --> MIDI", but I can't find this menu in plugdata.

I could get it to work in Reaper without configuring anything, but Live seems to behave differently.

This is probably a noob question, please forget my ignorance.

Thanks in advance.

**Comment by tomara-x** [comment]
yeah, that behavior is different across different daws, i'm not an ableton user but i think this might be helpful: https://help.ableton.com/hc/en-us/articles/209070189-Accessing-the-MIDI-output-of-a-VST-plug-in

**Reply by ericbhs** [reply]
Hello Tomara,
Thanks a lot for your anwser !
Unfortunately, as soon as I put the plugdata plugin on my Ableton Live rack, the "MDI TO" drop down menu turns into "AUDIO TO", so I can't access the midi output anymore...
Best regards

**Reply by tomara-x** [reply]
hello! booted windows and figured it out! (thanks for making me close all my tabs lol)
you need 2 midi tracks
1. add plugdata-fx on the first track (yeah the audio effect version)
2. i added a simple delay patch to test: (you can copy and paste it into plugdata)
```
#X obj 199 79 notein;
#X obj 179 149 pipe 250;
#X obj 250 149 pipe 250;
#X obj 190 237 noteout;
#X connect 0 0 1 0 empty;
#X connect 0 1 2 0 empty;
#X connect 1 0 3 0 empty;
#X connect 2 0 3 1 empty;
```
3. on the second track's "MIDI From" menu, select the track that has plugdata
4. put some midi instrument on this track to test
5. arm both tracks for recording
6. mash some keys, notes should be delayed from when you pressed the keys, meaning the effect is working

**Reply by ericbhs** [reply]
Wow, thanks a lot for your answer ! I'll test it back home tonight...
(and thanks for rebooting your computer for me ☺)

**Reply by ericbhs** [reply]
Working just fine, thanks a lot !

**Reply by tomara-x** [reply]
yaaay! have fun <3

**Reply by smutax** [reply]
Do you know if same applies to patches exported(compiled) as vst? I mean those designed to be "vst midi fx". Third party vst midi fx don't require the kind of workaround suggested above, right?

**Reply by tomara-x** [reply]
i have no experience with the compiler, so i don't know for sure. but my understanding is that vst3 (as a plugin format) was never intended to be used this way. so the way this is handled varies across daws. some will allow it, but others (like ableton) don't like it.
